### PR TITLE
Add Azure Pipeline for SonarCloud

### PR DESCRIPTION
**Pull request by tomasz-kaminski-sonarsource**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,3 +1,46 @@
+= C++ example project scanned on SonarCloud using Azure Pipelines
+// URIs:
+:uri-qg-status: https://sonarcloud.io/dashboard?id=sc-sq-c-family-examples_macos-cmake-azure-sc
+:img-qg-status: https://sonarcloud.io/api/project_badges/measure?project=sc-sq-c-family-examples_macos-cmake-azure-sc&metric=alert_status
+:uri-build-status: https://dev.azure.com/sc-sq-c-family-examples/macos-cmake-azure-sc/_build/latest?definitionId=2&branchName=azure-sc-analysis
+:img-build-status: https://dev.azure.com/sc-sq-c-family-examples/macos-cmake-azure-sc/_apis/build/status/sc-sq-c-family-examples.macos-cmake-azure-sc?branchName=azure-sc-analysis
```
Comment (arseniy-sonar) [inline README.adoc]:
```suggestion
:img-build-status: https://dev.azure.com/sc-sq-c-family-examples/macos-cmake-azure-sc/_apis/build/status/sc-sq-c-family-examples.macos-cmake-azure-sc
```

File: README.adoc [hunk 2]
```diff
@@ -1,3 +1,46 @@
+= C++ example project scanned on SonarCloud using Azure Pipelines
+// URIs:
+:uri-qg-status: https://sonarcloud.io/dashboard?id=sc-sq-c-family-examples_macos-cmake-azure-sc
+:img-qg-status: https://sonarcloud.io/api/project_badges/measure?project=sc-sq-c-family-examples_macos-cmake-azure-sc&metric=alert_status
+:uri-build-status: https://dev.azure.com/sc-sq-c-family-examples/macos-cmake-azure-sc/_build/latest?definitionId=2&branchName=azure-sc-analysis
```
Comment (arseniy-sonar) [inline README.adoc]:
```suggestion
:uri-build-status: https://dev.azure.com/sc-sq-c-family-examples/macos-cmake-azure-sc/_build/latest?definitionId=2
```

**Review by arseniy-sonar — CHANGES_REQUESTED** [review]
Looks good, just the stray branch in the build-status badge

File: README.adoc (hunk 1 above)
Comment (tomasz-kaminski-sonarsource) [reply README.adoc]: Removed all default branches

**Pull request merged**